src/main/java/org/jsoup/helper/HttpConnection.java

```diff
@@ -1239,58 +1239,58 @@ else if (needsMultipart(req)) {
 
         static void writePost(final HttpConnection.Request req, final OutputStream outputStream) throws IOException {
             final Collection<Connection.KeyVal> data = req.data();
-            final BufferedWriter w = new BufferedWriter(new OutputStreamWriter(outputStream, req.postDataCharset()));
             final String boundary = req.mimeBoundary;
 
-            if (boundary != null) { // a multipart post
-                for (Connection.KeyVal keyVal : data) {
+            try (BufferedWriter w = new BufferedWriter(new OutputStreamWriter(outputStream, req.postDataCharset()))) {
```

Comment:
The OutputStreamWriter is not closed -- the BufferedWriter's close does not cascade. I think we should close the OSW also, while we're at it.